### PR TITLE
Feature calendar page

### PR DESCRIPTION
**Pull request by guntzel-sebastian**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#Specific project name
```
Comment (JoeyKraemer) [inline .env:1]: This file shouldnt be here, delete it

File: .idea/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (JoeyKraemer) [inline .idea/.gitignore:1]: Are you using something out of the ".idea" folder if not please remove it

File: Calendar.php [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                if ($user = $userstmt->fetch(PDO::FETCH_OBJ)) {
+                    echo "<div class='WelcomeUser'><h2>Welcome," . $user . "</h2></div>";
+                }
+            } catch (Exception $e) {
+
+            }
+        }
+        ?>
+
+        <div id="calendar">
```
Comment (JoeyKraemer) [inline Calendar.php:44]: Check indentation

File: Calendar.php [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                }
+                catch(Exception $ex){
+                    echo "The following exception has occurred $ex";
+                }
+            }
+            else{
+                echo "No events planned";
+            }
+            ?>
+
```
Comment (JoeyKraemer) [inline Calendar.php:78]: remove empty line

File: Calendar.php [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    <table border='0'>
+        <?php
+        $date = getdate();
+
+        $mday = $date['mday'];
+        $mon = $date['mon'];
+        $wday = $date['wday'];
+        $month = $date['month'];
+        $year = $date['year'];
+
```
Comment (JoeyKraemer) [inline Calendar.php:90]: remove one empty line

File: Calendar.php [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            if (in_array( $d, $day_to_highlight))
+                $bg = "bg-green";
+            else
+                $bg = "bg-white";
+            // Highlights the current day
+            if($d == $mday)
+                echo("<td class='bg-blue'><a href='?event_date=$d' title='Detail of day'>$d</a></td>");
+            else
+                echo("<td class='$bg'><a href='?event_date=$d' title='Detail of day'>$d</a></td>");
+
```
Comment (JoeyKraemer) [inline Calendar.php:153]: rm one empty line

File: CalendarCSS.css [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+table {
+    font-family: arial, sans-serif;
+    border-spacing: 25px;
+    border-collapse: separate;
+    padding-left: 50px;
+    padding-right: 50px;
+    background-color: #FFFFFF;
+    display: block;
+    flex: auto;
+
```
Comment (JoeyKraemer) [inline CalendarCSS.css:21]: rm empty line

File: CalendarCSS.css [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+td, th {
+    text-align: left;
+    padding: 8px;
+    border-radius: 50%;
+    text-align: center;
+
```
Comment (JoeyKraemer) [inline CalendarCSS.css:29]: rm empty line

File: CalendarCSS.css [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    height: 25px;
+    width: 25px;
+    background-color: #dddddd;
+    color: #50595B;
+}
+
+a{
```
Comment (JoeyKraemer) [inline CalendarCSS.css:36]: space between "a" and "{" is missing, pleases look at the following as well

File: CalendarCSS.css [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+/*Hidden Elements*/
+
+/*IDs*/
+#container {
+    display: grid;
+    font-family: Inter;
+    font-weight: 400;
+    grid-template-columns: repeat(8,1fr);
+}
```
Comment (JoeyKraemer) [inline CalendarCSS.css:70]: add empty line

File: CalendarCSS.css [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+/*Classes*/
+.WelcomeUser {
+    grid-column: 2/8;
+    background-color: #F1F1F1;
+}
+
+.eventList{
+    padding-right: 10%;
+}
```
Comment (JoeyKraemer) [inline CalendarCSS.css:108]: add empty line at the end of the file

File: Calendar_demo.css [hunk 12]
```diff
@@ -0,0 +1,121 @@
+/*******************************Calendar Top Navigation*********************************/
```
Comment (JoeyKraemer) [inline Calendar_demo.css:1]: is it necessary to be part of this git?  If not please remove the files

File: docker-compose.yaml [hunk 13]
```diff
@@ -14,7 +14,6 @@ services:
         image: php:latest
```
Comment (JoeyKraemer) [inline docker-compose.yaml:14]: rm this file, it should not be part of this branch

File: calendarPHPBack.php [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+        echo("<td class='$bg'><a href='#' title='Detail of day'>$d</a></td>");
+
+
+    $startDay++;
+    if ($startDay > 6 && $d < $lastDay) {
+        $startDay = 0;
+        echo("</tr>");
+        echo("<tr>");
+    }
+    // <div id="date"><span style="font-size: 3em;" id="calendarDate">10</span><h3>January <br> 2023</h3></div>
```
Comment (JoeyKraemer) [inline calendarPHPBack.php:82]: rm commented code

**Review by JoeyKraemer — CHANGES_REQUESTED** [review]

**Review by YohanLoz — CHANGES_REQUESTED** [review]
okay, the calendar desperately needs to be able to switch months (i.e. look at februari when it's currently january). 
the comments need to be reworked (For multiple lines of code use  and only use // for single lines or comments). 
more comments are also required for things like "for ($d=1;$d<=$lastDay;$d++) {" because i have no idea what this does and especially not what it's for. 
also use 1 multi-line echo for large pieces of html instead of 10 seperate echo statements. Another thing is that the calendar should not have events where active = 0, (this should be easy to do, even with the current time constraints). 
max participants also seems to be absent from the data (this one is not that neccesary). 
whelp, good luck with all this.

**Comment by guntzel-sebastian** [comment]
Feedback has been implemented, only remaining issue is the User name function that would come from the login system.

**Review by JoeyKraemer — DISMISSED** [review]

**Review by JoeyKraemer — APPROVED** [review]

**Pull request merged**